File: dp_primer/dp_primer_apples_step1.cs
// 実行時間 20ms
using System;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] nab = Console.ReadLine().Split().Select( int.Parse ).ToArray();
        int[] p = new int[nab[0] + 4];

        if( nab[0] == 1 ) {
            Console.WriteLine( nab[1] );
            return;
        }
        p[0] = 0;
        p[1] = nab[1];
        foreach( int i in Range( 2, nab[0] - 1 ) ) {
            p[i] = Math.Min( p[i - 2] + nab[1], nab[2] * (int)Math.Ceiling( i / 5m ) );
        }
        Console.WriteLine( p[nab[0]] );
        return;
    }
}

File: dp_primer/dp_primer_apples_step1.py
# 実行時間 10ms
import math

def main():
    n, a, b = list(map(int, input().split()))
    d = [0, a]

    if n == 1:
        return a
    for i in range(2, n + 1):
        d.append(min(d[i - 2] + a, b * math.ceil(i / 5)))
    return d[i]

print(main())
